backend/src/services/comparisons/comparisons.service.ts

```diff
@@ -0,0 +1,194 @@
+      });
+
+    await prisma.comparisons_Complectations.create({
+      data: {
+        comparison_id: comparison.id,
+        complectation_id: complectationId,
+        position: carsInComparisonAmount + 1,
+      },
+    });
+
+    return comparison;
+  }
+
+  if (!activeComparison) {
+    const comparison = await prisma.comparison.create({
+      data: {
+        active: true,
+        created_at: new Date(),
+        updated_at: new Date(),
+        user_id: userId,
+      },
+    });
+
+    await prisma.comparisons_Complectations.create({
+      data: {
+        comparison_id: comparison.id,
+        complectation_id: complectationId,
+        position: 1,
+      },
+    });
+    return comparison;
+  }
+  return activeComparison;
+};
+
+const changeComparisonType = async ({
+  type,
+  userId,
+}: {
+  type: ComparisonType;
+  userId: string;
+}): Promise<Comparison> => {
+  const comparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+    select: {
+      id: true,
+    },
+  });
+  if (!comparison) {
+    throw new Error('There is no active comparison');
+  }
+  return await prisma.comparison.update({
+    where: { id: comparison.id },
+    data: { type },
+  });
+};
+
+const clearComparison = async (userId: string): Promise<Comparison> => {
+  const comparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+    select: {
+      id: true,
+    },
+  });
+  if (!comparison) {
+    throw new Error('There is no active comparison');
+  }
+  return await prisma.comparison.update({
+    where: { id: comparison.id },
+    data: { active: false },
+  });
+};
+
+const deleteCarFromComparison = async ({
+  complectationId,
+  userId,
+}: {
+  complectationId: string;
+  userId: string;
+}): Promise<void> => {
+  const comparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+    select: {
+      id: true,
+    },
+  });
+
+  if (!comparison) {
+    throw new Error('There is no active comparison');
+  }
+  await prisma.comparisons_Complectations.delete({
+    where: {
+      comparison_id_complectation_id: {
+        comparison_id: comparison.id,
+        complectation_id: complectationId,
+      },
+    },
+  });
+
+  const complectationsAmount = await prisma.comparisons_Complectations.count({
+    where: { comparison_id: comparison.id },
+  });
+
+  if (!complectationsAmount) {
+    await prisma.comparison.delete({
+      where: { id: comparison.id },
+    });
+  }
+};
+
+const getActiveComparisonCars = async (
+  userId: string,
+): Promise<(Complectation | null)[]> => {
+  const activeComparison = await prisma.comparison.findFirst({
+    where: {
+      active: true,
+      user_id: userId,
+    },
+    select: {
+      id: true,
+    },
+  });
+  if (!activeComparison) {
+    throw new Error('There is no active comparison');
```

Comment:
Maybe it will be better to return an empty array in case of no active comparison?  

If there isn't any, user still can visit comparison page and see nothing, so it looks like there isn't any reason to return  `400 Bad Request`.